clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1209,9 +1228,31 @@ void abortableError(RuntimeException e) {
             transitionToAbortableError(e);
         }
 
+        /**
+         * Determines if an error should be treated as abortable or fatal, based on transaction state and configuration.
+         *
+         *  - **Transactional with Epoch Bumping Available**:
+         *      If the producer can trigger an epoch bump, the error is treated as abortable,
+         *      allowing recovery by restarting the transaction with a new epoch.
+         *
+         * - **TransactionV2 Enabled**:
+         *      When TransactionV2 is enabled, the producer automatically bumps the epoch on each transaction commit or abort,
+         *      making errors abortable without manual intervention.
+         *
+         * - **Non-Transactional Producers**:
+         *      For non-transactional producers, errors are considered abortable since no transaction rollback is necessary.
```

Comment:
Hmm. This comment is a little confusing. I didn't think that idempotent producers had abortable errors.

Reply:
Oya that's an oversight on my part, since I added the condition !isTransactional and in these cases we set epochBumpRequired true and treat it as an abortable error I got confused

Reply:
Should we remove the !isTransactional condition too in this case?